### PR TITLE
Implement `@behaviour` for `Routes.Repo`

### PR DESCRIPTION
**Pull request by alicia-mbta**
#### Summary of changes
**Asana Ticket:** [Define a RoutesApi behaviour for V3Api.Routes](https://app.asana.com/0/555089885850811/1189396929591660)

### CODE REVIEW

**Comment by alicia-mbta** [comment]
This draft PR is an attempt to achieve what was discussed in https://github.com/mbta/dotcom/pull/665

File: apps/routes/lib/repo.ex [hunk 1]
```diff
@@ -212,7 +215,7 @@ defmodule Routes.Repo do
   @doc """
   The Green Line.
   """
```
Comment (paulswartz) [inline apps/routes/lib/repo.ex:217]: Does it make sense to move the documentation into `Routes.RepoApi` as well?

Comment (alicia-mbta) [reply apps/routes/lib/repo.ex:217]: You're right, I'll re-organize the documentation.

File: apps/routes/lib/repo.ex [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 
   @default_opts [include: "route_patterns"]
 
-  @doc """
-
-  Returns a list of all the routes
-
-  """
-  @spec all() :: [Route.t()]
+  @impl Routes.RepoApi
```
Comment (thecristen) [inline apps/routes/lib/repo.ex:17]: Do you need both the `@behavior` and the `@impl` declarations?

Comment (paulswartz) [reply apps/routes/lib/repo.ex:17]: `@impl` tells the compiler which interface this is a function for. It'll help you out by failing if you forget any implementations.

**Review by thecristen — APPROVED** [review]
👍

File: apps/routes/lib/route.ex [hunk 3]
```diff
@@ -202,6 +202,29 @@ defmodule Routes.Route do
 
   def silver_line, do: @silver_line
 
+  @doc """
+  Determines if the given route data is hidden
+  """
+  @spec route_hidden?(%{id: String.t()}) :: boolean
+  def route_hidden?(%{id: "746"}), do: true
```
Comment (paulswartz) [inline apps/routes/lib/route.ex:209]: Since you're on `Route` already, this can be `hidden?()`

File: apps/routes/lib/route.ex [hunk 4]
```diff
@@ -202,6 +202,29 @@ defmodule Routes.Route do
 
   def silver_line, do: @silver_line
 
+  @doc """
+  Determines if the given route data is hidden
+  """
+  @spec hidden?(%{id: String.t()}) :: boolean
```
Comment (paulswartz) [inline apps/routes/lib/route.ex:208]: `hidden?(t()) :: boolean`

File: apps/routes/lib/repo_api.ex [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  @doc """
+  Given a route pattern, returns the list of routes matching it.
+  """
+  @callback by_stop_with_route_pattern(String.t()) :: [Route.t()]
+
+  @doc """
+  Parses json into a list of routes, or an error if it happened.
+  """
+  @callback handle_response(JsonApi.t() | {:error, any}) :: {:ok, [Route.t()]} | {:error, any}
```
Comment (paulswartz) [inline apps/routes/lib/repo_api.ex:51]: This looks like it's an implementation detail of the real `RouteApi`: I don't think a fake implementation would need to implement it?

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**